front/lib/resources/plan_resource.ts

```diff
@@ -0,0 +1,182 @@
+import type {
+  Attributes,
+  CreationAttributes,
+  ModelStatic,
+  Transaction,
+} from "sequelize";
+
+import type { Authenticator } from "@app/lib/auth";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import type { Subscription } from "@app/lib/resources/storage/models/plans";
+import { PlanModel } from "@app/lib/resources/storage/models/plans";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+import type { ResourceFindOptions } from "@app/lib/resources/types";
+import type { PlanType, Result } from "@app/types";
+import { Ok } from "@app/types";
+
+export type PlanAttributes = Omit<
+  Attributes<PlanModel>,
+  "id" | "createdAt" | "updatedAt"
+>;
+
+// These limits are applied to all plans during the trial period.
+export const TRIAL_LIMITS: Partial<PlanAttributes> = {
+  maxUsersInWorkspace: 5,
+  maxMessages: 100,
+  maxMessagesTimeframe: "day",
+};
+
+// This function is used to get the trial version of a plan.
+export function getTrialVersionForPlan(plan: PlanModel): PlanModel {
+  return PlanModel.build({
+    ...plan.get(),
+    ...TRIAL_LIMITS,
+  });
+}
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unsafe-declaration-merging
+export interface PlanResource extends ReadonlyAttributesType<PlanModel> {}
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class PlanResource extends BaseResource<PlanModel> {
+  static model: ModelStatic<PlanModel> = PlanModel;
+
+  constructor(model: ModelStatic<PlanModel>, blob: Attributes<PlanModel>) {
+    super(PlanModel, blob);
+  }
+
+  //build a new plan resource from a given plan model. It is not saved to the database.
```

Comment:
`nit`
We usually structure comments this way:
```suggestion
  // Build a new plan resource from a given plan model. It is not saved to the database.
```
And max 80-100 cols.